python/lsst/ts/externalscripts/auxtel/latiss_wep_align.py

```diff
@@ -0,0 +1,286 @@
+                wep_results,
+            ) = await loop.run_in_executor(
+                pool,
+                functools.partial(
+                    run_wep,
+                    self.intra_visit_id,
+                    self.extra_visit_id,
+                    2 * self.side,
+                    self.timeout_get_image,
+                ),
+            )
+
+            # output from wep are in microns, need to convert to nm.
+            self.zern = [
+                -wep_results.outputZernikesAvg[0][4] * 1e3,
+                wep_results.outputZernikesAvg[0][3] * 1e3,
+                wep_results.outputZernikesAvg[0][0] * 1e3,
+            ]
+
+        return self.calculate_results()
+
+
+def run_wep(
+    intra_visit_id: int,
+    extra_visit_id: int,
+    donut_diameter: int,
+    timeout_get_image: float,
+) -> typing.Tuple[Struct, Struct, Struct]:
+
+    best_effort_isr = BestEffortIsr()
+
+    exposure_intra = get_image(
+        parse_visit_id(intra_visit_id),
+        best_effort_isr,
+        timeout=timeout_get_image,
+    )
+    exposure_extra = get_image(
+        parse_visit_id(extra_visit_id),
+        best_effort_isr,
+        timeout=timeout_get_image,
+    )
+
+    quick_frame_measurement_config = QuickFrameMeasurementTask.ConfigClass()
+    quick_frame_measurement_task = QuickFrameMeasurementTask(
+        config=quick_frame_measurement_config
+    )
+
+    result_intra = quick_frame_measurement_task.run(
+        exposure_intra, donutDiameter=donut_diameter
+    )
+    result_extra = quick_frame_measurement_task.run(
+        exposure_extra, donutDiameter=donut_diameter
+    )
+
+    if not result_intra.success or not result_extra.success:
+        raise RuntimeError(
+            f"Centroid finding algorithm was unsuccessful. "
+            f"Intra image ({exposure_intra}) success is {result_intra.success}. "
+            f"Extra image ({exposure_extra}) success is {result_extra.success}."
+        )
+
+    dy = (
+        result_extra.brightestObjCentroidCofM[0]
+        - result_intra.brightestObjCentroidCofM[0]
+    )
+    dx = (
+        result_extra.brightestObjCentroidCofM[1]
+        - result_intra.brightestObjCentroidCofM[1]
+    )
+    dr = np.sqrt(dy**2 + dx**2)
+
+    position_out_of_range = dr > 100
+
+    donut_catalog_intra = get_donut_catalog(result_intra, exposure_intra.getWcs())
+    donut_catalog_extra = get_donut_catalog(
+        *(
+            (result_extra, exposure_extra.getWcs())
+            if position_out_of_range
+            else (result_intra, exposure_intra.getWcs())
+        )
+    )
+
+    config = EstimateZernikesLatissTaskConfig()
+    config.donutStampSize = donut_diameter
+    config.donutTemplateSize = donut_diameter
+    config.opticalModel = "onAxis"
+
+    task = EstimateZernikesLatissTask(config=config)
+
+    camera = best_effort_isr.butler.get(
+        "camera",
+        dataId={"instrument": "LATISS"},
+        collections="LATISS/calib/unbounded",
```

Comment:
what is this collection?
I guess i'm a little confused what is going on here. This grabs information about the camera that is needed by the  `EstimateZernikesLatissTask` method?
The reason you don't just put it inside the class is why? Speed? The class is already latiss specific...

Reply:
This is running on a separate process, so it does not have access to the properties of the class. I am not sure what that collection is as well.